src/Models/Services/Utils.ts

```diff
@@ -43,12 +43,67 @@ const parser = createParser(ModelParsingOption.PermitAnyTopLevelElement);
 
 /** Parse DTDL models via model parser */
 export const parseDTDLModelsAsync = async (dtdlInterfaces: DtdlInterface[]) => {
-    const modelDict = await parser.parse(
-        dtdlInterfaces.map((dtdlInterface) => JSON.stringify(dtdlInterface))
-    );
+    let modelDict = null;
+    try {
+        modelDict = await parser.parse(
+            dtdlInterfaces.map((dtdlInterface) => JSON.stringify(dtdlInterface))
+        );
+    } catch (e) {
+        /* Some models failed to parse, we will try to remove them and parse the rest of the models */
+        logDtdlParserError(e);
+        try {
+            const parsingErrorCauses = e?._parsingErrors.map(
+                (pe) => pe?.cause
+            ) as Array<string>;
+
+            /* The parsing error cause is not a model ID, but it contains a model ID 
+               so we remove the a model if its ID is contained in the cause */
+            const modelsWithoutParserErrors = dtdlInterfaces.filter((intf) => {
+                const modelId = intf['@id'];
+                return !parsingErrorCauses.filter((parseError) =>
+                    parseError.includes(modelId)
+                ).length;
+            });
+
+            // Recursively parse models if we successfully removed some models, and there are still some left
+            if (
+                modelsWithoutParserErrors.length < dtdlInterfaces.length &&
+                modelsWithoutParserErrors.length > 0
+            ) {
+                console.log('Removing model that failed to parse and retrying');
+                return await parseDTDLModelsAsync(modelsWithoutParserErrors);
```

Comment:
do we want to keep track of some kind of depth so we can bust out if we get too far in?

Reply:
yup, added depth of 20